tests/components/binary_sensor/test_tod.py

```diff
@@ -0,0 +1,839 @@
+"""Test Times of the Day Binary Sensor."""
+import unittest
+from unittest.mock import patch
+from datetime import timedelta, datetime
+import pytz
+
+from homeassistant import setup
+import homeassistant.core as ha
+from homeassistant.const import STATE_OFF, STATE_ON
+import homeassistant.util.dt as dt_util
+from homeassistant.setup import setup_component
+from tests.common import (
+    get_test_home_assistant, assert_setup_component)
+from homeassistant.helpers.sun import (
+    get_astral_event_date, get_astral_event_next)
+
+
+class TestBinarySensorTod(unittest.TestCase):
```

Comment:
We prefer standalone pytest test functions for new tests.

Reply:
@MartinHjelmare Is that a problem now?  I missed that requirement in developer's guide. I was following the Template Sensor implementation. How could I know it's wrong? Do I need to take any action?

Reply:
You're welcome to rewrite the tests to pytest test functions if you want. It's not required at this point since this PR is merged.

I commented so that both contributors and reviewers can take note for the future.

Our developers docs are not complete.